### PR TITLE
Add flag to skip cla/codecov failure in fast forward

### PR DESCRIPTION
**Pull request by yutongz**
https://github.com/istio/istio-testing/issues/157

### CODE REVIEW

File: tools/github_helper/main.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ var (
 	base        = flag.String("base", "stable", "The base branch used for PR.")
 	head        = flag.String("head", "master", "The head branch used for PR.")
 	pullRequest = flag.Int("pr", 0, "The Pull request to use.")
+	codecov     = flag.Bool("codecov", false, "Skip code coverage failure")
```
Comment (sebastienvas) [inline tools/github_helper/main.go]: Please create a unique flag for those that takes in a comma separated list of context to skip.

File: tools/github_helper/main.go [hunk 2]
```diff
@@ -38,12 +40,14 @@ type ghConst struct {
...
-	Base   string
-	Head   string
-	Pr     int
-	Client *github.Client
+	Owner   string
+	Repo    string
+	Base    string
+	Head    string
+	Pr      int
+	Codecov bool
```
Comment (sebastienvas) [inline tools/github_helper/main.go]: Change this with something like ChecksToSkip

**Review by sebastienvas — CHANGES_REQUESTED** [review]
PLease create a unique flag.

File: tools/github_helper/main.go [hunk 3]
```diff
@@ -284,12 +281,20 @@ func (h helper) updatePullRequest(pr *github.PullRequest, s *github.CombinedStat
...
-	if state == GH.failure && (h.Codecov || h.Cla) {
+	if state == GH.failure && (len(h.CheckToSkip) > 0) {
 		privilege := true
 		for _, status := range s.Statuses {
 			if *status.State != GH.success {
-				if !(h.Codecov && strings.HasPrefix(*status.Context, "codecov")) && !(h.Cla && strings.HasPrefix(*status.Context, "cla")) {
-					//Failed, and not eligible for any privilege
+				skip := false
+				for _, check := range h.CheckToSkip {
+					if *status.Context == check {
```
Comment (sebastienvas) [inline tools/github_helper/main.go]: Have you tried this ? Should we use some kind of regex instead ?

**Review by sebastienvas — COMMENTED** [review]
Just a minor comment

File: tools/github_helper/main.go (hunk 3 above)
Comment (yutongz) [reply tools/github_helper/main.go]: I tested its logic by creating a fake combinedStatus directly in the updatePullRequest(), working but naive.
And does this pattern work for you:
keyWord := "cla"   //input flag
pattern := fmt.Sprintf("(^|/)%s(/|$)", keyWord)

Comment (sebastienvas) [reply tools/github_helper/main.go]: what about fmt.Sprintf(".\*%s.\*", keyWord) ?

Comment (yutongz) [reply tools/github_helper/main.go]: Well, I think it depends on what possible contexts we have. Are we going to have some confused cases. But in this situation, I think we may prefer false negative to false positive. If we can skip one check while the program block it, well, it's easy to manually check and pass it. On contrary, we may miss a really failure.

**Review by sebastienvas — DISMISSED** [review]

**Comment by sebastienvas** [comment]
To plug it to Jenkins, please update the following to take a build parameters for exceptions 
https://github.com/istio/istio-testing/blob/master/src/org/istio/testutils/Utilities.groovy#L99
And update https://testing.istio.io/job/istio-testing/job/verify-stable-prs/

**Comment by yutongz** [comment]
PTAL

**Review by sebastienvas — APPROVED** [review]
Awesome

**Pull request merged**